sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobAsyncClient.java

```diff
@@ -359,6 +368,27 @@ as we can guarantee we only need at most two buffers for any call to write (two
         }
     }
 
+    private Mono<Response<BlockBlobItem>> determineUploadFullOrChunked(final Flux<ByteBuffer> data,
+        final Function<Flux<ByteBuffer>, Mono<Response<BlockBlobItem>>> path1,
+        final BiFunction<Flux<ByteBuffer>, Long, Mono<Response<BlockBlobItem>>> path2) {
+        final int[] predicateCount = { 4 * Constants.MB };
```

Comment:
Should this use the blockSize specified in `ParallelTransferOptions` instead of hardcoding this value?

Reply:
I think that is a reasonable change, basically have the logic of if we cannot fill an entire buffer just send it as a single upload.

Reply:
@rickle-msft Thoughts on this change, maybe hold off on it until the possible extension of `ParallelTransferOptions` to include a property that indicates limit to trigger Put Blob vs Stage Blocks and put Block List.

Reply:
I'm ok on holding off on this because we actually want this feature for uploadFromFile as well. We can do both at once.